backends/utils/history_server.go

```diff
@@ -0,0 +1,526 @@
+		request.Proto.MaxStartTime = "now"
+	}
+	startTime, err := utils.Str2unixTime(request.Proto.MinStartTime)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse 'start_time' filter")
+	}
+	endTime, err := utils.Str2unixTime(request.Proto.MaxStartTime)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse 'end_time' filter")
+	}
+
+	container, err := m.createV3ioClient(request.Proto.Session, request.Password.Get(), request.Token.Get())
+	if err != nil {
+		return err
+	}
+
+	filter := historyFilter{Backend: request.Proto.Backend,
+		Action:       request.Proto.Action,
+		User:         request.Proto.User,
+		Table:        request.Proto.Table,
+		Container:    request.Proto.Container,
+		MinStartTime: startTime,
+		MaxStartTime: endTime,
+		MinDuration:  request.Proto.MinDuration,
+		MaxDuration:  request.Proto.MaxDuration}
+
+	iter, err := utils.NewAsyncItemsCursor(container,
+		&v3io.GetItemsInput{Path: m.LogsFolderPath + "/", AttributeNames: []string{"__name"}},
+		8, nil, m.logger)
+
+	if err != nil {
+		return fmt.Errorf("Failed to list Frames History log folder %v for read, err: %v", m.LogsFolderPath, err)
+	}
+
+	i := 1
+
+	// Create default constant columns. Other type-specific column will be added during iteration
+	var userNameColData, backendNameColData, tableNameColData, actionTypeColData, containerColData []string
+	var actionDurationColData []int64
+	var startTimeColData []time.Time
+	var nullColumns []*pb.NullValuesMap
+	var hasNullsInCurrentDF bool
+	customColumnsByName := make(map[string][]string)
+	var indexColData []int
+
+	// go over all log files in the folder
+	for iter.Next() {
+		currentFilePath := path.Join(m.LogsFolderPath, iter.GetField("__name").(string))
+		lineIterator, err := v3ioutils.NewFileContentLineIterator(currentFilePath, m.MaxBytesInNginxRequest, container)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to get '%v'", currentFilePath)
+		}
+
+		var rowsInCurrentDF int
+		for lineIterator.Next() {
+			if rowsInCurrentDF == maxLogsInMessage {
+
+				frame, err := createDF(userNameColData, backendNameColData, tableNameColData, actionTypeColData, containerColData,
+					actionDurationColData, startTimeColData,
+					indexColData, customColumnsByName, nullColumns, hasNullsInCurrentDF)
+				if err != nil {
+					return err
+				}
+				out <- frame
+
+				rowsInCurrentDF = 0
+				userNameColData = []string{}
+				backendNameColData = []string{}
+				tableNameColData = []string{}
+				actionDurationColData = []int64{}
+				startTimeColData = []time.Time{}
+				actionTypeColData = []string{}
+				containerColData = []string{}
+				indexColData = []int{}
+
+				customColumnsByName = make(map[string][]string)
+				nullColumns = []*pb.NullValuesMap{}
+				hasNullsInCurrentDF = false
+			}
+
+			var entry HistoryEntry
+
+			if err := json.Unmarshal(lineIterator.At(), &entry); err != nil {
+				return fmt.Errorf("error reading logs. json marshal error:%v", err)
```

Comment:
Better: `failed to parse json: %v`. Even better, include the log that failed the parser.